src/adapters/http/index.ts

```diff
@@ -0,0 +1,109 @@
+import Adapter from '../../lib/adapter.js';
+import GleeMessage from '../../lib/message.js';
+import http from 'http';
+import { validateData } from '../../lib/util.js';
+import GleeError from '../../errors/glee-error.js';
+import * as url from 'url';
+
+class HttpAdapter extends Adapter {
+    private res: any
+
+    name(): string {
+        return 'HTTP server'
+    }
+
+    async connect(): Promise<this> {
+        return this._connect();
+    }
+
+    async send(message: GleeMessage): Promise<void> {
+        console.log('control reached send');
+        return this._send(message);
+    }
+
+    _connect(): Promise<this> {
+        return new Promise((resolve, reject) => {
+            const serverUrl = new URL(this.serverUrlExpanded);
+            const httpServer = this.glee.options?.websocket?.httpServer || http.createServer();
+            const asyncapiServerPort = serverUrl.port || 80;
+            const optionsPort = this.glee.options?.websocket?.port;
+            const port = optionsPort || asyncapiServerPort;
+
+            httpServer.on('request', (req, res) => {
+                this.res = res;
+                let { pathname } = new URL(req.url, serverUrl);
+
+                if (pathname.startsWith('/')) {
+                    pathname = pathname.substring(1);
+                }
+
+                if (!this.parsedAsyncAPI.channel(pathname)) {
+                    res.end('HTTP/1.1 404 Not Found\r\n\r\n');
+                    const err = new Error(`A client attempted to connect to channel ${pathname} but this channel is not defined in your AsyncAPI file.`)
+                    this.emit('error', err)
+                    return reject(err)
+                }
+
+                const {query: searchParams} = url.parse(req.url, true);
+                const httpChannelBinding = this.parsedAsyncAPI.channel(pathname).binding('http');
+
+                if (httpChannelBinding) {
+                    const { query, method } = httpChannelBinding;
+
+                    if(method) {
+                        if(req.method !== method) {
+                            this.emit('error', new Error(`${req.method} is not allowed for this url`));
+                            res.end('HTTP/1/1 400 Bad Request\r\n\r\n');
+                            return;
+                        }
+                    }
+
+                    if (query) {
+                        const {isValid, humanReadableError, errors} = validateData(searchParams, query);
+                        if(!isValid){
+                            const err = new GleeError({humanReadableError, errors});
+                            console.log('ERROR', err);
+                            this.emit('error', err);
+                            res.end('HTTP/1.1 400 Bad Request\r\n\r\n');
+                            return;
+                        }
+                    }
+                }
```

Comment:
I am trying to validate the query parameters but getting an issue. All the query parameters are being treated as a string so if my query defined in `asyncapi.yaml` as a string then whatever the `query` is validation results in positive. 

```
bindings:
      http:
        bindingVersion: 0.1.0
        type: request
        method: GET
        query:
          type: object
          properties:
            name:
              type: string

```

for this `?name=shreak`, `?name` and `?name=12` get validated positively which is something that shouldn't happen.

Reply:
🤔  I think you have to try to find the type on purpose. It's true that query parameters are always treated as strings so you'll only get positives in this case. You can probably do this: try `Number(query.name)` and if you don't get `NaN` it means it's a number so you can safely do `query.name = Number(query.name)`. Same with `null`. If you get that `query.name` is `undefined` or `null` then you can safely assign it to `null`.

That said, you don't check for `query.name` but for each key in the object. Don't forget about array values. Hope that helps :)